sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/UnitTestResultsImportSensor.java

```diff
@@ -20,75 +20,59 @@
 package org.sonar.plugins.dotnet.tests;
 
 import java.io.File;
-import org.sonar.api.batch.bootstrap.ProjectDefinition;
-import org.sonar.api.batch.sensor.Sensor;
 import org.sonar.api.batch.sensor.SensorContext;
 import org.sonar.api.batch.sensor.SensorDescriptor;
 import org.sonar.api.measures.CoreMetrics;
 import org.sonar.api.notifications.AnalysisWarnings;
+import org.sonar.api.scanner.sensor.ProjectSensor;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 import org.sonarsource.dotnet.shared.plugins.DotNetPluginMetadata;
 
 /**
  * This class is responsible to handle all the C# and VB.NET unit test results reports (parse and report back to SonarQube).
  */
-public class UnitTestResultsImportSensor implements Sensor {
+public class UnitTestResultsImportSensor implements ProjectSensor {
 
   private static final Logger LOG = Loggers.get(UnitTestResultsImportSensor.class);
 
   private final WildcardPatternFileProvider wildcardPatternFileProvider = new WildcardPatternFileProvider(new File("."), File.separator);
   private final UnitTestResultsAggregator unitTestResultsAggregator;
-  private final ProjectDefinition projectDef;
```

Comment:
for my education: what was this replaced with?

Reply:
I see it was only being used in
```
    if (projectDef.getParent() == null) {
      analyze(context, new UnitTestResults());
    }
```
but I don't know what that was for

Reply:
Aaa, it was because it was ran at all modules, and it should only do something at the top-level module that didn't have any parent.

Reply:
Exactly. The sensor was looking for the last top-level module to run on. And that's now implemented by inheriting `ProjectSensor`. So this workaround is not needed anymore and that's nice.